mne_bids/path.py

```diff
@@ -443,6 +497,17 @@ def _find_matching_sidecar(bids_fname, bids_root, suffix, allow_fail=False):
         and no sidecar_fname was found
 
     """
+    if isinstance(bids_fname, str):
+        params = parse_bids_filename(bids_fname)
+        bids_fname = BIDSPath(subject=params.get('sub'),
+                              session=params.get('ses'),
+                              task=params.get('task'),
+                              acquisition=params.get('acq'),
+                              run=params.get('run'),
+                              processing=params.get('proc'),
+                              recording=params.get('rec'),
+                              space=params.get('space'))
+
```

Comment:
Needed to add this to make sure the passed in object gets turned into a `BIDSPath`.

Reply:
I think we should change the API of this function. If I have a BIDSPath to begin with, I don't want to have to pass the `bids_root` to this function manually… Can we make it **require** a BIDSPath and drop the `bids_root` kwarg? Then you could also get rid of this entire block here.

Reply:
Hmmm... I'm taking a look rn, and if we do that, then do we just change `read_raw_bids` and `write_raw_bids` too?

where do we stop?

Or is this fine, since it's a private function.

Reply:
Jk it works fine.